http/backup_service.go

```diff
@@ -106,6 +131,73 @@ func (h *BackupHandler) handleBackupShard(w http.ResponseWriter, r *http.Request
 	}
 }
 
+func (h *BackupHandler) handleBackupMetadata(w http.ResponseWriter, r *http.Request) {
+	span, r := tracing.ExtractFromHTTPRequest(r, "BackupHandler.handleBackupMetadata")
+	defer span.Finish()
+
+	ctx := r.Context()
+
+	// Lock the sqlite and bolt databases prior to writing the response to prevent
+	// data inconsistencies.
+	h.BackupService.LockKVStore()
+	defer h.BackupService.UnlockKVStore()
+
+	h.SqlBackupRestoreService.LockSqlStore()
```

Comment:
Is there a chance we could end up w/ a deadlock here where the KV store is locked & waiting on sql, and some other process has sql locked & is waiting on KV?

Reply:
I don't think so - at least not right now. The only other thing that uses the sql database is the notebooks service. Notebooks will get the write lock when modifying a notebook. The only way the modification of a notebook should interact with the KV is reading data from it during the request to get permissions info. This requires getting a read lock, and the KV store is locked here with a read lock as well. There currently isn't anything you can do with the sql database that would require a write to the KV. Except for backups, I guess!

So I can't think of a scenario where there would be a deadlock, really due to the fact that the two databases are almost completely decoupled. At least for notebooks...and annotations should work the same way. If we start doing things which do writes to both databases simultaneously I could see it being a problem. Then there could be a deadlock if the external request starts right between these two locks. So...hopefully we don't ever do that 😸.